From fb65ac2e3028c4fc160a47448bc44d73d282c9f2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Caol=C3=A1n=20McNamara?= <caolanm@redhat.com>
Date: Mon, 8 Jun 2015 12:21:01 +0100
Subject: [PATCH] Resolves: tdf#49407 enable CaseMap property in impress/draw

I don't see why this was explicitly disabled, works fine
out of the box for me, import/export already hooked up
in odf

Change-Id: I5e6bdbc6a4f3cbcd97330c7d9fb33589489afee2
---
 cui/source/inc/chardlg.hxx      | 10 ----------
 sd/sdi/drtxtob.sdi              |  6 ++++++
 sd/source/core/typemap.cxx      |  1 +
 sd/source/ui/dlg/dlgchar.cxx    |  1 -
 sd/source/ui/dlg/prltempl.cxx   |  1 -
 sd/source/ui/dlg/tabtempl.cxx   |  1 -
 sd/source/ui/func/fuchar.cxx    |  1 +
 sd/source/ui/func/futext.cxx    |  7 ++++---
 sd/source/ui/view/drtxtob.cxx   |  3 ++-
 sd/source/ui/view/drviews2.cxx  | 11 +++++++++++
 sd/source/ui/view/drviewsf.cxx  |  1 +
 sw/source/ui/chrdlg/chardlg.cxx |  8 +-------
 12 files changed, 27 insertions(+), 24 deletions(-)

diff --git a/cui/source/inc/chardlg.hxx b/cui/source/inc/chardlg.hxx
index 4eb009a..a023a45 100644
--- a/cui/source/inc/chardlg.hxx
+++ b/cui/source/inc/chardlg.hxx
@@ -32,16 +32,6 @@
 class SvxFontListItem;
 class FontList;
 
-
-
-#define DISABLE_CASEMAP             ((sal_uInt16)0x0001)
-#define DISABLE_WORDLINE            ((sal_uInt16)0x0002)
-#define DISABLE_BLINK               ((sal_uInt16)0x0004)
-#define DISABLE_UNDERLINE_COLOR     ((sal_uInt16)0x0008)
-
-#define DISABLE_LANGUAGE            ((sal_uInt16)0x0010)
-#define DISABLE_HIDE_LANGUAGE       ((sal_uInt16)0x0020)
-
 // class SvxCharBasePage -------------------------------------------------
 
 class SvxCharBasePage : public SfxTabPage
diff --git a/sd/sdi/drtxtob.sdi b/sd/sdi/drtxtob.sdi
index 3e975d4..ed564e4 100644
--- a/sd/sdi/drtxtob.sdi
+++ b/sd/sdi/drtxtob.sdi
@@ -92,6 +92,12 @@ shell TextObjectBar
         StateMethod = GetAttrState;
     ]
 
+    SID_ATTR_CHAR_CASEMAP   // ole : ?, status : ?
+    [
+        ExecMethod = Execute;
+        StateMethod = GetAttrState;
+    ]
+
     SID_ATTR_PARA_ADJUST_LEFT  // ole : ?, status : ?
     [
         ExecMethod = Execute;
diff --git a/sd/source/core/typemap.cxx b/sd/source/core/typemap.cxx
index a5548a3..65f7cf0 100644
--- a/sd/source/core/typemap.cxx
+++ b/sd/source/core/typemap.cxx
@@ -21,6 +21,7 @@
 
 #include <editeng/outliner.hxx>
 #include <editeng/eeitem.hxx>
+#include <editeng/cmapitem.hxx>
 #include <editeng/langitem.hxx>
 #include <editeng/fhgtitem.hxx>
 #include <editeng/colritem.hxx>
diff --git a/sd/source/ui/dlg/dlgchar.cxx b/sd/source/ui/dlg/dlgchar.cxx
index a23cd80..5e44899 100644
--- a/sd/source/ui/dlg/dlgchar.cxx
+++ b/sd/source/ui/dlg/dlgchar.cxx
@@ -60,7 +60,6 @@ void SdCharDlg::PageCreated( sal_uInt16 nId, SfxTabPage &rPage )
     }
     else if (nId == mnCharEffects)
     {
-        aSet.Put (SfxUInt16Item(SID_DISABLE_CTL,DISABLE_CASEMAP));
         rPage.PageCreated(aSet);
     }
 }
diff --git a/sd/source/ui/dlg/prltempl.cxx b/sd/source/ui/dlg/prltempl.cxx
index 74c010e..c503c93 100644
--- a/sd/source/ui/dlg/prltempl.cxx
+++ b/sd/source/ui/dlg/prltempl.cxx
@@ -281,7 +281,6 @@ void SdPresLayoutTemplateDlg::PageCreated( sal_uInt16 nId, SfxTabPage &rPage )
 
     else if (nId == mnEffects)
         {
-            aSet.Put (SfxUInt16Item(SID_DISABLE_CTL,DISABLE_CASEMAP));
             rPage.PageCreated(aSet);
         }
 }
diff --git a/sd/source/ui/dlg/tabtempl.cxx b/sd/source/ui/dlg/tabtempl.cxx
index 61514ee..1189922 100644
--- a/sd/source/ui/dlg/tabtempl.cxx
+++ b/sd/source/ui/dlg/tabtempl.cxx
@@ -152,7 +152,6 @@ void SdTabTemplateDlg::PageCreated( sal_uInt16 nId, SfxTabPage &rPage )
     }
     else if (nId == m_nFontEffectId)
     {
-        aSet.Put (SfxUInt16Item(SID_DISABLE_CTL,DISABLE_CASEMAP));
         rPage.PageCreated(aSet);
     }
     else if (nId == m_nTextId)
diff --git a/sd/source/ui/func/fuchar.cxx b/sd/source/ui/func/fuchar.cxx
index 174391a..b8166c2 100644
--- a/sd/source/ui/func/fuchar.cxx
+++ b/sd/source/ui/func/fuchar.cxx
@@ -106,6 +106,7 @@ void FuChar::DoExecute( SfxRequest& rReq )
                     SID_ATTR_CHAR_FONTHEIGHT,
                     SID_ATTR_CHAR_COLOR,
                     SID_ATTR_CHAR_KERNING,
+                    SID_ATTR_CHAR_CASEMAP,
                     SID_SET_SUPER_SCRIPT,
                     SID_SET_SUB_SCRIPT,
                     0 };
diff --git a/sd/source/ui/func/futext.cxx b/sd/source/ui/func/futext.cxx
index ac6fb71..33e5339 100644
--- a/sd/source/ui/func/futext.cxx
+++ b/sd/source/ui/func/futext.cxx
@@ -85,12 +85,13 @@ static sal_uInt16 SidArray[] = {
     SID_ATTR_CHAR_FONT,               //   10007
     SID_ATTR_CHAR_POSTURE,            //   10008
     SID_ATTR_CHAR_WEIGHT,             //   10009
-    SID_ATTR_CHAR_SHADOWED,     //10010
-    SID_ATTR_CHAR_STRIKEOUT,        //10013
+    SID_ATTR_CHAR_SHADOWED,           //   10010
+    SID_ATTR_CHAR_STRIKEOUT,          //   10013
     SID_ATTR_CHAR_UNDERLINE,          //   10014
     SID_ATTR_CHAR_FONTHEIGHT,         //   10015
     SID_ATTR_CHAR_COLOR,              //   10017
-    SID_ATTR_CHAR_KERNING,          //10018
+    SID_ATTR_CHAR_KERNING,            //   10018
+    SID_ATTR_CHAR_CASEMAP,            //   10019
     SID_ATTR_PARA_ADJUST_LEFT,        //   10028
     SID_ATTR_PARA_ADJUST_RIGHT,       //   10029
     SID_ATTR_PARA_ADJUST_CENTER,      //   10030
diff --git a/sd/source/ui/view/drtxtob.cxx b/sd/source/ui/view/drtxtob.cxx
index 25913dd..45c868f 100644
--- a/sd/source/ui/view/drtxtob.cxx
+++ b/sd/source/ui/view/drtxtob.cxx
@@ -171,8 +171,9 @@ void TextObjectBar::GetAttrState( SfxItemSet& rSet )
             case SID_ATTR_CHAR_FONTHEIGHT:
             case SID_ATTR_CHAR_WEIGHT:
             case SID_ATTR_CHAR_POSTURE:
-        case SID_ATTR_CHAR_SHADOWED:
+            case SID_ATTR_CHAR_SHADOWED:
             case SID_ATTR_CHAR_STRIKEOUT:
+            case SID_ATTR_CHAR_CASEMAP:
             {
                 sal_uInt16 stretchX = 100;
                 sal_uInt16 stretchY = 100;
diff --git a/sd/source/ui/view/drviews2.cxx b/sd/source/ui/view/drviews2.cxx
index d4ed8fa..ffbb503 100644
--- a/sd/source/ui/view/drviews2.cxx
+++ b/sd/source/ui/view/drviews2.cxx
@@ -87,6 +87,7 @@
 #include <vcl/svapp.hxx>
 #include <vcl/waitobj.hxx>
 
+#include <editeng/cmapitem.hxx>
 #include <editeng/escapementitem.hxx>
 #include <editeng/kernitem.hxx>
 #include <editeng/wghtitem.hxx>
@@ -3032,6 +3033,16 @@ void DrawViewShell::ExecChar( SfxRequest &rReq )
             }
         }
         break;
+    case SID_ATTR_CHAR_CASEMAP:
+        if( rReq.GetArgs() )
+        {
+            SFX_REQUEST_ARG( rReq, pItem, SvxCaseMapItem, SID_ATTR_CHAR_CASEMAP , false );
+            if (pItem)
+            {
+                aNewAttr.Put(*pItem);
+            }
+        }
+        break;
     case SID_SET_SUB_SCRIPT:
         {
             SvxEscapementItem aItem( EE_CHAR_ESCAPEMENT );
diff --git a/sd/source/ui/view/drviewsf.cxx b/sd/source/ui/view/drviewsf.cxx
index 56d7a88..a5183c1 100644
--- a/sd/source/ui/view/drviewsf.cxx
+++ b/sd/source/ui/view/drviewsf.cxx
@@ -409,6 +409,7 @@ void DrawViewShell::GetAttrState( SfxItemSet& rSet )
             case SID_ATTR_CHAR_WEIGHT:
             case SID_ATTR_CHAR_COLOR:
             case SID_ATTR_CHAR_KERNING:
+            case SID_ATTR_CHAR_CASEMAP:
             case SID_SET_SUB_SCRIPT:
             case SID_SET_SUPER_SCRIPT:
             {
diff --git a/sw/source/ui/chrdlg/chardlg.cxx b/sw/source/ui/chrdlg/chardlg.cxx
index b20dfdc2..2483fda 100644
--- a/sw/source/ui/chrdlg/chardlg.cxx
+++ b/sw/source/ui/chrdlg/chardlg.cxx
@@ -112,13 +112,7 @@ void SwCharDlg::PageCreated( sal_uInt16 nId, SfxTabPage &rPage )
     }
     else if (nId == m_nCharExtId)
     {
-        if(m_nDialogMode == DLG_CHAR_DRAW || m_nDialogMode == DLG_CHAR_ANN)
-            aSet.Put (SfxUInt16Item(SID_DISABLE_CTL,DISABLE_CASEMAP));
-
-        else
-        {
-            aSet.Put (SfxUInt32Item(SID_FLAG_TYPE,SVX_PREVIEW_CHARACTER|SVX_ENABLE_FLASH));
-        }
+        aSet.Put (SfxUInt32Item(SID_FLAG_TYPE,SVX_PREVIEW_CHARACTER|SVX_ENABLE_FLASH));
         rPage.PageCreated(aSet);
     }
     else if (nId == m_nCharPosId)
-- 
2.4.0

